### PR TITLE
feat: add grpc support for recommender service

### PR DESCRIPTION
**Pull request by raymondfeng**
This is a spin-off from https://github.com/strongloop/loopback4-example-shopping/pull/268.

1. Allow recommender service to expose gRPC endpoints
2. Allow shopping to connect to recommender over gRPC and REST

### CODE REVIEW

File: packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts [hunk 1]
```diff
@@ -238,21 +241,33 @@ describe('UserController', () => {
...
         .get(`/users/${newUser.id}/recommend`)
-        .expect(200, recommendations);
+        .expect(200, recommendations['user001']);
+    });
+
+    it('returns product recommendations for a user using gRPC', async () => {
+      const recommender = await app.get<RecommenderService>(
+        'services.RecommenderGrpcService',
+      );
+      const products = await recommender.getProductRecommendations('user001');
```
Comment (agnes512) [inline packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts]: Can we have a test case for the user not in recommendations such as `user003`?

File: packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts [hunk 2]
```diff
@@ -238,21 +241,33 @@ describe('UserController', () => {
 
   describe('user product recommendation (service) api', () => {
     let recommendationService: HttpServer;
+    let recommendationGRPCService: Server;
 
     before(async () => {
       recommendationService = createRecommendationServer();
       await recommendationService.start();
+      recommendationGRPCService = createGRPCRecommendationServer();
+      recommendationGRPCService.start();
```
Comment (agnes512) [inline packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts]: Do we need `await` for this line as well?

Comment (raymondfeng) [reply packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts]: No. This start does not return a promise.

File: packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts (hunk 1 above)
Comment (raymondfeng) [reply packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts]: The recommender is a mock service that returns same products for users other than `user002`

Comment (raymondfeng) [reply packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts]: I added one for user004.

**Pull request merged**